app/src/main/java/com/example/userinfo/MainActivity.kt

```diff
@@ -0,0 +1,182 @@
+package com.example.userinfo
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import java.util.regex.Pattern
+
+class MainActivity : AppCompatActivity() {
+    val pincodePattern = Pattern.compile("^[1-9][0-9]{5}$")
+    val emailPattern=Pattern.compile("^[A-za-z0-9_.-]+@[a-z]+\\.+(com|co.in)")
+    val userNamePattern=Pattern.compile("^[A-za-z0-9 ]*")
+    val phoneNumberPattern=Pattern.compile("^[1-9][0-9]{9}")
+    var visibility=true
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        val userName = findViewById<EditText>(R.id.userNameEt)
+        val email = findViewById<EditText>(R.id.emailEt)
+        val number = findViewById<EditText>(R.id.numberEt)
+        val pincode = findViewById<EditText>(R.id.pincodeEt)
+        val address = findViewById<EditText>(R.id.addressEt)
+        val validate = findViewById<Button>(R.id.buttonLabel)
+        val cancel=findViewById<Button>(R.id.cancelButton)
+        val confirm=findViewById<Button>(R.id.confirmButton)
+
+        validate.setOnClickListener() {
+            val isAllFieldsChecked = checkAllFields(userName,email,number,pincode,address)
+            if (isAllFieldsChecked) {
+                val validateResult=validateAllFields(userName,email,number,pincode,address)
+                if (validateResult) {
+                    visibility=false
+                    modifyVisibility(userName,email,number,pincode,address,validate,cancel,confirm,visibility)
+
+                }
+            }
+            confirm.setOnClickListener() {
```

Comment:
We can setOnClickListener while creating only, if `confirm` button is part of layout